scripts/android_tracy_tutorial.sh

```diff
@@ -0,0 +1,245 @@
+  TRACY_DEPS_COUNT=5
+  TRACY_DEPS_INSTALLED="$(apt list $TRACY_DEPS) 2>/dev/null | grep installed | wc -l"
+  if [ $TRACY_DEPS_INSTALLED != $TRACY_DEPS_COUNT ]
+  then
+    print_status "Installing dependencies now - assuming Debian."
+    sudo apt install $TRACY_DEPS
+  fi
+  make -C "${TRACY_ROOT}/profiler/build/unix" -j12 release
+fi
+
+echo
+
+print_status "Ensuring that we have the input MLIR file..."
+if [ ! -f "${IREE_INPUT_MLIR}" ]
+then
+  print_status "Set IREE_INPUT_MLIR to point to some iree input MLIR file. Not found at current value ${IREE_INPUT_MLIR}."
+  if [ "${IREE_INPUT_MLIR}" == "${DEFAULT_IREE_INPUT_MLIR}" ]
+  then
+    print_status "Okay, we actually know how to generate that file, ${IREE_INPUT_MLIR}, but it will take a while."
+    print_status "Press the return key to continue..."
+    pushd "${IREE_ROOT}"
+    scripts/get_e2e_artifacts.py --test_suites=mobile_bert_squad_tests
+    popd
+  fi
+fi
+
+if [ ! -f "${IREE_INPUT_MLIR}" ]
+then
+  print_status "FATAL: we should have ${IREE_INPUT_MLIR} by that point."
+  exit 1
+fi
+
+echo
+
+print_status "Building IREE for Android in ${IREE_BUILD_ANDROID}..."
+
+mkdir -p "${IREE_BUILD_ANDROID}"
+pushd "${IREE_BUILD_ANDROID}"
+
+cmake -G Ninja ../iree \
+  -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
+  -DANDROID_ABI="arm64-v8a" \
+  -DANDROID_PLATFORM=android-30 \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DIREE_BUILD_COMPILER=OFF \
+  -DIREE_BUILD_SAMPLES=OFF  \
+  -DIREE_HOST_C_COMPILER=`which "$CC"` \
+  -DIREE_HOST_CXX_COMPILER=`which "$CXX"` \
+  -DIREE_ENABLE_RUNTIME_TRACING=ON
+
+cmake --build . --target iree_tools_iree-translate
+
+popd
+echo
+
+print_status "Compiling the input MLIR file into a IREE module..."
+
+IREE_COMPILED_MODULE=/tmp/android_module.vmfb
+IREE_LOG=/tmp/iree-translate.log
+
+rm -rf "${IREE_COMPILED_MODULE}"
+
+IREE_LLVMAOT_LINKER_PATH="${IREE_LLVMAOT_LINKER_PATH}" \
+  "${IREE_BUILD_ANDROID}/host/bin/iree-translate" \
+    --iree-hal-target-backends=dylib-llvm-aot \
+    --iree-mlir-to-vm-bytecode-module \
+    --iree-llvm-target-triple=aarch64-linux-android \
+    /tmp/iree/modules/MobileBertSquad/iree_input.mlir \
+    -o "${IREE_COMPILED_MODULE}" \
+    2>"${IREE_LOG}"
+
+if [ ! -f "${IREE_COMPILED_MODULE}" ]
+then
+  print_status "iree-translate failed to produce ${IREE_COMPILED_MODULE}. Log saved in ${IREE_LOG}. First few lines:"
+  # The whole log might be enormous if it contains a big MLIR dump.
+  head -n10 "${IREE_LOG}"
+  print_status "tip: check if IREE_LLVMAOT_LINKER_PATH was correctly set."
+  exit 1
+fi
+
+echo
+
+DEVICE_IREE_COMPILED_MODULE=/data/local/tmp/android_module.vmfb
+
+print_status "Pushing the compiled module to the device..."
+adb push "${IREE_COMPILED_MODULE}" "${DEVICE_IREE_COMPILED_MODULE}"
+echo
+
+print_status "Pushing the IREE benchmarking program to the device..."
+adb push "${IREE_BUILD_ANDROID}"/iree/tools/iree-benchmark-module /data/local/tmp
```

Comment:
Does this also need to be built (see the `cmake --build . --target <targets>` above)?

Reply:
d'oh! thanks!